dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -58,64 +58,16 @@ static const columninfo RecycleBinColumns[] =
 
 #define COLUMNS_COUNT  6
 
-/*
- * Recycle Bin folder
- */
-
-HRESULT CRecyclerExtractIcon_CreateInstance(LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
+static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl)
 {
-    CComPtr<IDefaultExtractIconInit> initIcon;
-    HRESULT hr = SHCreateDefaultExtractIcon(IID_PPV_ARG(IDefaultExtractIconInit, &initIcon));
-    if (FAILED_UNEXPECTEDLY(hr))
-        return hr;
+    LPPIDLDATA pdata = _ILGetDataPointer(pidl);
 
-    /* FIXME: This is completely unimplemented */
-    initIcon->SetNormalIcon(swShell32Name, 0);
+    if (pdata && pdata->type == 0x00)
+        return (PIDLRecycleStruct*) & (pdata->u.crecycle);
 
-    return initIcon->QueryInterface(riid, ppvOut);
+    return NULL;
 }
 
-class CRecycleBinEnum :
```

Comment:
Why are you moving stuff around? It makes it hard to review the diff.

Reply:
It will be more spagetti code the need of putting a declaration here and then the body in the original position

Reply:
And what's more this is only used on this file, so it's not worth adding the declaration to the header file

Reply:
What I'm asking is -- why do you now need a forward declaration? It's not obvious to me what necessitates it.

Reply:
Please refrain from moving code around if there is no need for it,
mixing code cleanup / implementation in one PR is undesirable.

Reply:
Ok then i can revert this

Reply:
if this is the only change needed

Reply:
Done :D